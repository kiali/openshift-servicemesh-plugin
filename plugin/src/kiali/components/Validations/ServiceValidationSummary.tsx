import * as React from 'react';
import { Text, TextVariants } from '@patternfly/react-core';
import { ValidationSummary } from './ValidationSummary';
import { kialiStyle } from 'styles/StyleUtils';

const tooltipListStyle = kialiStyle({
  textAlign: 'left',
  border: 0,
  padding: '0 0 0 0',
  margin: '0 0 0 0'
});

export class ServiceValidationSummary extends ValidationSummary {
  tooltipSummary() {
    return (
      <>
        <Text style={{ textAlign: 'left', textEmphasis: 'strong' }} component={TextVariants.p}>
          Service validation result
        </Text>
        <div className={tooltipListStyle}>
          {this.severitySummary().map(cat => (
            <div key={cat}>{cat}</div>
          ))}
        </div>
      </>
    );
  }

  tooltipContent() {
    return this.tooltipSummary();
  }
}
