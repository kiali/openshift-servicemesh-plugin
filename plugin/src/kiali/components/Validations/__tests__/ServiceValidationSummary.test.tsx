import * as React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { ServiceValidationSummary } from '../ServiceValidationSummary';

const testScenario = (summary: any) => {
  const wrapper = shallow(summary);
  expect(shallowToJson(wrapper)).toBeDefined();
  expect(shallowToJson(wrapper)).toMatchSnapshot();
};

describe('ServiceValidationSummary renders', () => {
  it('success icon when all components are valid', () => {
    testScenario(<ServiceValidationSummary id={'1'} errors={0} warnings={0} />);
  });

  it('warning icon when all components are valid', () => {
    testScenario(<ServiceValidationSummary id={'2'} errors={0} warnings={1} />);
    testScenario(<ServiceValidationSummary id={'2'} errors={0} warnings={2} />);
  });

  it('error icon when all components are valid', () => {
    testScenario(<ServiceValidationSummary id={'3'} errors={1} warnings={0} />);
    testScenario(<ServiceValidationSummary id={'3'} errors={1} warnings={2} />);
  });
});
