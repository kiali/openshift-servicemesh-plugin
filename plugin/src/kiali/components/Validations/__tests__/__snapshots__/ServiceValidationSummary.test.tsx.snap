// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceValidationSummary renders error icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Service validation result
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 error found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="error"
  />
</Tooltip>
`;

exports[`ServiceValidationSummary renders error icon when all components are valid 2`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Service validation result
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 error found
        </div>
        <div>
          2 warnings found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="error"
  />
</Tooltip>
`;

exports[`ServiceValidationSummary renders success icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Service validation result
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          No issues found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="correct"
  />
</Tooltip>
`;

exports[`ServiceValidationSummary renders warning icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Service validation result
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 warning found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="warning"
  />
</Tooltip>
`;

exports[`ServiceValidationSummary renders warning icon when all components are valid 2`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Service validation result
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          2 warnings found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="warning"
  />
</Tooltip>
`;
